localstack/services/iam/resource_providers/aws_iam_role.py

```diff
@@ -0,0 +1,245 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+import json
+from pathlib import Path
+from typing import Optional, Type, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    CloudFormationResourceProviderPlugin,
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+from localstack.utils.functions import call_safe
+
+
+class IAMRoleProperties(TypedDict):
+    AssumeRolePolicyDocument: Optional[dict | str]
+    Arn: Optional[str]
+    Description: Optional[str]
+    ManagedPolicyArns: Optional[list[str]]
+    MaxSessionDuration: Optional[int]
+    Path: Optional[str]
+    PermissionsBoundary: Optional[str]
+    Policies: Optional[list[Policy]]
+    RoleId: Optional[str]
+    RoleName: Optional[str]
+    Tags: Optional[list[Tag]]
+
+
+class Policy(TypedDict):
+    PolicyDocument: Optional[str | dict]
+    PolicyName: Optional[str]
+
+
+class Tag(TypedDict):
+    Key: Optional[str]
+    Value: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
```

Comment:
minor: worth moving this into `localstack.services.cloudformation.resource_provider`?

Reply:
Probably not since we want the files to be standalone for packaging purposes. We could add a different isolated file though that wraps the REPEATED_INVOCATION thing as we discussed automatically and add it to the request object that the handlers receive.